analyzers/tests/SonarAnalyzer.UnitTest/Rules/ExpressionComplexityTest.cs

```diff
@@ -34,6 +34,53 @@ public class ExpressionComplexityTest
                 .WithOptions(ParseOptionsHelper.FromCSharp8)
                 .Verify();
 
+        [DataTestMethod]
+        [DataRow("==")]
+        [DataRow("!=")]
+        [DataRow("<")]
+        [DataRow("<=")]
+        [DataRow(">")]
+        [DataRow(">=")]
+        public void ExpressionComplexity_TransparentComparissionOperators(string @operator) =>
+            builderCS.AddSnippet($$$"""
+            class C
+            {
+                public void M()
+                {
+                    var x = true && true && (1 {{{@operator}}} (true ? 1 : 1));         // Compliant (Make sure, the @operator is not increasing complexity)
+                    var y = true && true && true && (1 {{{@operator}}} (true ? 1 : 1)); // Noncompliant {{Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).}}
+                }
+            }
+            """)
+            .Verify();
+
+        [DataTestMethod]
+        [DataRow("o", "??")]
+        [DataRow("i", "|")]
+        [DataRow("i", "^")]
+        [DataRow("i", "&")]
```

Comment:
Bitwise `|` and `&` are controversial. You could also consider them complexity increasing. The old rule treated those as complexity-increasing. I'm not sure about the other operators but I would assume these were increasing complexity too. See e.g. 

https://peach.sonarsource.com/code?id=dotnet-runtime&selected=dotnet-runtime%3Asrc%2Flibraries%2FMicrosoft.CSharp%2Fsrc%2FMicrosoft%2FCSharp%2FRuntimeBinder%2FSymbolTable.cs&line=117

https://peach.sonarsource.com/code?id=dnspy&selected=dnspy%3AdnSpy%2FdnSpy.Contracts.Debugger%2FBreakpoints%2FModules%2FDbgModuleBreakpointSettings.cs&line=99
which was marked as an issue before.